dcos/httpclient.go

```diff
@@ -0,0 +1,69 @@
+package dcos
```

Comment:
please don't call this package `dcos` and, instead, pick a more specific name. perhaps `dcosclient` or `client`.

naming a package `dcos` is akin to naming a package `util` or `common` - it provides very little (i.e. too broad) contextual value for the things declared within it.

if this package is going to get more complex, please consider sub-packages for very closely related types.

Reply:
@jdef this would be a matter of design. The previous version of this was based on sub-packages but we explicitly wanted a flat design. 

In general this is not carved in stone yet we still can move these again into subpackages. But the new concept we spoke about in our meeting with the CLI was the flat design oriented on github.com/google/go-github 

However lets try to cover this in the design doc. IMO it shouldn't stop us from having the first methods and structs implemented.

Reply:
about `dcos` vs `dcosclient` or `client` 

`client` - this is way to genereal. Having a `client.NewClient` is a bit weird. 
`dcosclient` - I see what you're trying with it but to me it doesnt add value as `dcosclient.Client` is still a duplicate.

`dcos.Client` is exactly what we're trying to achieve here.

`dcos.NewClient().Marathon.NewApp()` would be an example for the way we're trying to implement this.

the great benefit this delivers is that even packages that are not focussed but broadly used could be adopted with this.

```go
d := dcos.NewClient()
c := chronos.New(chronosURL, d.HTTPClient)
```

Reply:
> please don't call this package `dcos` and, instead, pick a more specific name. perhaps `dcosclient` or `client`.
> 
> naming a package `dcos` is akin to naming a package `util` or `common` - it provides very little (i.e. too broad) contextual value for the things declared within it.

I don't think `dcos` is akin to `util`, especially not from the user's perspective. Reading `dcos.Client` gives you all the context you need, I think, and I'd say that it's more helpful than `client.Client` (and while `dcosclient.Client` gives the same amount of info, it's redundant).

Since this library is meant to be used a client SDK in other software I don't really think it's too broad.

> if this package is going to get more complex, please consider sub-packages for very closely related types.

👍 Will keep an eye on this! I pushed for using a flat hierarchy for now, which I think allows for easier development until patterns emerge and I'd rather have them emerge than to shoehorn something into a package-layout